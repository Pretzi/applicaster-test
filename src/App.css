.App {
  display: flex;
  justify-content: center;
}

.App__container {
  display: flex;
  justify-content: center;
  width: 500px;
  margin: 40px 15px 0 15px;
  flex-direction: column;
}

.App__title {
  font-weight: bold;
  margin-bottom: 25px;
}

.pt-label {
  font-weight: bold;
}

.App__gif-container {
  display: flex;
  flex-direction: column;
  align-items: center;
}