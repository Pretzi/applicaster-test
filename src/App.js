import React, { Component } from 'react';
import { FormGroup, Button, Spinner } from "@blueprintjs/core";
import AppContainer from './containers/AppContainer';
import Input from './components/common/input/Input';
import RandomGifs from './components/random-gifs/RandomGifs';
import PropTypes from 'prop-types';

import './App.css';

class App extends Component {
  constructor(props) {
    super(props);

    const {
      category,
      timer,
      numberOfImages
    } = this.props;

    this.state = {
      category,
      timer,
      numberOfImages
    };

    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleInputChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  handleSubmit(e) {
    e.preventDefault();

    const {
      category,
      numberOfImages,
      timer
    } = this.state;

    this.props.fetchGifs(
      category,
      Number(numberOfImages),
      timer
    );
  }

  render() {
    const {
      gifs,
      showNumberOfImages,
      showTimer,
      loading,
      notFound
    } = this.props

    return (
      <div className="App">
        <div className="App__container">
          <h2 className="App__title">Search Random Gifs</h2>
          <form
            className="App__form"
            onSubmit={this.handleSubmit}
          >
            <FormGroup label="Gif Category">
              <Input
                value={this.state.category}
                onChange={e => this.handleInputChange(e)}
                placeholder="Tacos"
                type="string"
                name="category"
                icon="pt-icon-tag"
              />
            </FormGroup>

            {showNumberOfImages &&
              <FormGroup label="Number Of Images">
                <Input
                  value={this.state.numberOfImages}
                  onChange={e => this.handleInputChange(e)}
                  placeholder="3"
                  name="numberOfImages"
                  type="number"
                  icon="pt-icon-media"
                />
              </FormGroup>
            }

            {showTimer &&
              <FormGroup label="Timer" helperText="Time is in seconds">
                <Input
                  value={this.state.timer}
                  onChange={e => this.handleInputChange(e)}
                  placeholder="10 seconds"
                  name="timer"
                  type="number"
                  icon="pt-icon-time"
                />
              </FormGroup>
            }

            <Button
              intent="PRIMARY"
              text="Search"
              large
              onClick={this.handleSubmit}
              type="submit"
            />
          </form>

          <div className="App__gif-container">
            <RandomGifs gifs={gifs} />
            {loading && <Spinner />}
            {notFound &&
              <span className="App__not-found">Gifs not found</span>
            }
          </div>
        </div>
      </div>
    );
  }
}

App.displayName = 'App';
App.propTypes = {
  fetchGifs: PropTypes.func
};
App.defaultProps = {
  category: '',
  numberOfImages: 3,
  timer: 10,
  showNumberOfImages: true,
  showTimer: true,  
};

export default AppContainer(App);

