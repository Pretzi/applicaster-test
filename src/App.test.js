import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

describe('<App />', () => {
  it('Should render as expected', () => {
    const wrapper = shallow(
      <App />
    );
    expect(wrapper).toMatchSnapshot();
  })
});