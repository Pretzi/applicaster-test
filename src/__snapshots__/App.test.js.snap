// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should render as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="App__container"
>
        <h2
                className="App__title"
        >
                Search Random Gifs
        </h2>
        <FormGroup
                label="Gif Category"
        >
                <Input
                        name="category"
                        onChange={[Function]}
                        placeholder="Tacos"
                        type="search"
                        value=""
                />
        </FormGroup>
</div>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="App__title"
>
            Search Random Gifs
</h2>,
          <FormGroup
            label="Gif Category"
>
            <Input
                        name="category"
                        onChange={[Function]}
                        placeholder="Tacos"
                        type="search"
                        value=""
            />
</FormGroup>,
        ],
        "className": "App__container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search Random Gifs",
            "className": "App__title",
          },
          "ref": null,
          "rendered": "Search Random Gifs",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Input
              name="category"
              onChange={[Function]}
              placeholder="Tacos"
              type="search"
              value=""
/>,
            "label": "Gif Category",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "category",
              "onChange": [Function],
              "placeholder": "Tacos",
              "type": "search",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="App__container"
>
          <h2
                    className="App__title"
          >
                    Search Random Gifs
          </h2>
          <FormGroup
                    label="Gif Category"
          >
                    <Input
                              name="category"
                              onChange={[Function]}
                              placeholder="Tacos"
                              type="search"
                              value=""
                    />
          </FormGroup>
</div>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="App__title"
>
              Search Random Gifs
</h2>,
            <FormGroup
              label="Gif Category"
>
              <Input
                            name="category"
                            onChange={[Function]}
                            placeholder="Tacos"
                            type="search"
                            value=""
              />
</FormGroup>,
          ],
          "className": "App__container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search Random Gifs",
              "className": "App__title",
            },
            "ref": null,
            "rendered": "Search Random Gifs",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Input
                name="category"
                onChange={[Function]}
                placeholder="Tacos"
                type="search"
                value=""
/>,
              "label": "Gif Category",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "category",
                "onChange": [Function],
                "placeholder": "Tacos",
                "type": "search",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
